Update changelog 3.6 to clearly requiring applications which depend on etcd 3.6 packages must be built with go version >= 1.18

ahrtr opened the pull request:
etcdserver starts to using generic in https://github.com/etcd-io/etcd/pull/14515, so any applications which depend on etcdserver must be built with go version >= 1.18.

Signed-off-by: Benjamin Wang <wachao@vmware.com>


Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

cc @serathius @spzala @ptabor

ptabor commented:
How about similifying the message ?
 - any package in etcd-3.6 requires go 1.18 (as it's pretty likely we will take some 1.18 dependency soon)
  - if user want's to use go 1.17 with etcd-3.6, they still can link to older client (v3.5) that still will connect to etcd-3.6

ptabor commented:
There is Benjamin's answer to my comment that I missposted in the other thread: 
https://github.com/etcd-io/etcd/pull/14515#issuecomment-1268526845

spzala reviewed (commented):
@ahrtr thanks and good idea to update changelog! One comment I have is the sentence `because etcdserver starts using generic.` looks incomplete and may not be read clearly. I would suggest to modify it to `because etcdserver now use generic b-tree.`

ahrtr commented:
Thanks both @ptabor and @spzala for the comment. I just updated the PR per the discussion in https://github.com/etcd-io/etcd/pull/14515 to use simplified message. I added more detailed info in the commit message.

spzala reviewed (approved):
This looks better @ahrtr thanks! One nit if you want to make change is to change `3.6` to `v3.6` and `1.18` to `v1.18` as just two line above we mention versions as `etcd v3.6 will not run on v3.7`.  Keeping it consistence is a good reading experience for users.
